packages/core/src/sync-historical/index.ts

```diff
@@ -535,13 +536,32 @@ export const createHistoricalSync = async (
       }),
     ).then((traces) => traces.flat());
 
-    if (traces.length > 0) {
-      await args.syncStore.insertTraces({
-        traces,
+    if (isKilled) return;
+
+    await args.syncStore.insertTraces({
+      traces,
+      chainId: args.network.chainId,
+    });
+
+    if (isKilled) return;
+
+    if (shouldGetTransactionReceipt(filter)) {
+      const transactionHashes = new Set(
+        traces.map(({ transaction }) => transaction.hash),
+      );
+
+      const transactionReceipts = await Promise.all(
+        Array.from(transactionHashes).map((hash) =>
+          _eth_getTransactionReceipt(args.requestQueue, { hash }),
+        ),
+      );
```

Comment:
Definitely a follow-up concern, but I'm guessing we could de-duplicate these requests across filters. And, if we add support for `eth_getBlockReceipts` that would also be suitable for reuse (same way we do it for transaction inputs).